crates/test-collector/src/lib.rs

```diff
@@ -338,59 +338,58 @@ fn extract_fork_config_from_args(db: &dyn SyntaxGroup, attr: &Attribute) -> Opti
         return None;
     }
 
-    let block_id_type = block_id
-        .path(db)
-        .elements(db)
-        .last()
-        .unwrap()
-        .identifier(db)
-        .to_string();
+    let block_id_type = elements[1].clone();
 
     let args = block_id.arguments(db).args(db).elements(db);
     let expr = match args.first()?.arg_clause(db) {
         ArgClause::Unnamed(unnamed_arg_clause) => Some(unnamed_arg_clause.value(db)),
         _ => None,
     }?;
-    let block_id = try_get_block_id(db, &block_id_type, &expr)?;
+    let block_id_value = try_get_block_id(db, &block_id_type, &expr)?;
 
-    Some(RawForkConfig::Params(RawForkParams { url, block_id }))
+    Some(RawForkConfig::Params(RawForkParams {
+        url,
+        block_id_type,
+        block_id_value,
+    }))
 }
 
-fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<BlockId> {
+fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<String> {
     match block_id_type {
         "Number" => {
             if let Expr::Literal(value) = expr {
-                return Some(BlockId::Number(
-                    u64::try_from(value.numeric_value(db).unwrap()).ok()?,
-                ));
+                return Some(
+                    u64::try_from(value.numeric_value(db).unwrap())
+                        .ok()?
+                        .to_string(),
+                );
             }
-            None
         }
         "Hash" => {
+            // TODO #1179: add range check
             if let Expr::Literal(value) = expr {
-                return Some(BlockId::Hash(
-                    Felt252::from(value.numeric_value(db).unwrap()).into_(),
-                ));
+                return Some(value.numeric_value(db).unwrap().to_string());
             }
-            None
         }
         "Tag" => {
             if let Expr::Path(block_tag) = expr {
-                let tag = block_tag
-                    .elements(db)
-                    .last()
-                    .unwrap()
-                    .identifier(db)
-                    .to_string();
-                return match tag.as_str() {
-                    "Latest" => Some(BlockId::Tag(BlockTag::Latest)),
-                    _ => None,
-                };
+                let tag_elements = block_tag.elements(db);
+                if tag_path_is_valid(&tag_elements, db) {
+                    return Some("Latest".to_string());
+                }
             }
-            None
         }
-        _ => None,
-    }
+        _ => (),
+    };
+
+    None
+}
+
+// Only valid options are `BlockTag::Latest` and `Latest`
+fn tag_path_is_valid(tag_elements: &[PathSegment], db: &dyn SyntaxGroup) -> bool {
+    (tag_elements.len() == 1
+        || (tag_elements.len() == 2 && tag_elements[0].identifier(db).as_str() == "BlockTag"))
+        && tag_elements.last().unwrap().identifier(db).as_str() == "Latest"
 }
```

Comment:
Why there are two valid options here?

Reply:
We pretend that the stuff enclosed in attributes are type instances e.g. `BlockId::Tag(BlockTag::Latest)` but in fact we never even check if the types are present in the scope, if they are valid etc., we just treat path parts as strings. We should prob fix that someday, but when it was introduces no one really reviewed it carefully I guess (myself included)
cc @war-in

Reply:
Assume mostly backwards compatibilty. Even our codebase uses `BlockId::Tag(Latest)` instead of `BlockId::Tag(BlockTag::Latest)` which I'd assume is the correct syntax.

Reply:
We allow using both `Latest` and `BlockTag::Latest` syntax for `BlockId::Tag`. Technically only `BlockTag::Latest` should be allowed type wise, but just `Latest` is already used extensively in multiple places.

Reply:
Aren't those values produces by our code? Why we produce two different values?

Reply:
These are not produced by use. User provides them when creating a forking test with `#[fork(...)]` attribute. We just validate them here.